### PR TITLE
Add `npm run diff`

### PR DESCRIPTION
**Pull request by saschanaz**
A checklist to help your pull request get merged faster:
- [x] Summarize your changes
- [x] Link to related issues or pull requests, if any: #6863 

This intends to succeed what @foolip did in #6862 using git commands. Example:

```
> npm run diff ef6b236 80991e0

> @mdn/browser-compat-data@2.0.7 diff
> node scripts/diff.js "ef6b236" "80991e0"

api.Document.caretRangeFromPoint:
 → support.safari_ios.version_added is edited (5 → 4)
api.Document.contentType:
 → support.safari.version_added is edited (9.1 → 9)
 → support.safari_ios.version_added is edited (9.3 → 9)
api.Document.createNodeIterator:
 → support.safari_ios.version_added is edited (true → 1)
api.Document.createTouch:
 → support.safari_ios.version_added is edited (true → ≤3)
api.Document.createTouchList:
 → support.safari_ios.version_added is edited (true → ≤3)
api.Document.currentScript:
 → support.safari.version_added is edited (7 → 6.1)
 → support.safari_ios.version_added is edited (7 → 8)
api.Document.head:
 → support.safari_ios.version_added is edited (true → 4)
api.Document.hidden:
 → support.safari.version_added is edited (7 → 6.1)
api.Document.onpointerlockchange:
 → support.safari_ios.version_added is edited (10.3 → false)
api.Document.onpointerlockerror:
 → support.safari_ios.version_added is edited (10.3 → false)
api.Document.onreadystatechange:
 → support.safari_ios.version_added is edited (6 → 5)
api.HTMLFrameSetElement:
 → status.deprecated is edited (false → true)
api.HTMLFrameSetElement.event_handlers:
 → status.deprecated is edited (false → true)
api.HTMLFrameSetElement.onlanguage:
 → status.deprecated is edited (false → true)
api.HTMLFrameSetElement.onstorage:
 → status.deprecated is edited (false → true)
api.NavigatorConcurrentHardware:
 → support.safari.version_added is edited (null → false)
 → support.safari_ios.version_added is edited (null → false)
api.NavigatorConcurrentHardware.hardwareConcurrency:
 → support.safari.version_added is edited (null → false)
 → support.safari_ios.version_added is edited (null → false)
api.NavigatorConcurrentHardware.worker_support:
 → support.safari.version_added is edited (null → false)
 → support.safari_ios.version_added is edited (null → false)
css.selectors:
 → css.selectors.-moz-ui-invalid is deleted
 → css.selectors.target-within is added
css.selectors.user-invalid:
 → Newly added
```

`npm run diff` is a shorthand of `npm run diff origin/HEAD`, which compares current changes against `origin/HEAD`. The longhand is primarily for debugging and test.

### CODE REVIEW

**Comment by queengooborg** [comment]
Hey @saschanaz, this is LGTM for the most part and I'd love to add it in!  Would you mind rebasing this real quick?

**Comment by saschanaz** [comment]
Done, but I only have npm 7. Feel free to regenerate package-lock.json.

**Comment by queengooborg** [comment]
We upgraded to use NodeJS v16, which uses NPM v8 and package-lock v2.  Looks like this one needs another quick rebase!

**Review by queengooborg — APPROVED** [review]
This script is LGTM and I'm excited to integrate it!  I'd like @foolip's review as well though, since this builds upon his work!

File: package.json [hunk 1]
```diff
@@ -34,6 +34,7 @@
     "browser-specs": "~3.10.0",
     "chalk": "~4.1.0",
     "compare-versions": "~4.1.1",
+    "deep-diff": "^1.0.2",
```
Comment (queengooborg) [inline package.json:37]:
```suggestion
    "deep-diff": "~1.0.2",
```

**Review by queengooborg — CHANGES_REQUESTED** [review]
Ah, happened to notice a small thing -- could we use `~` vs. `^` for the version number to maintain consistency between the dependencies?

**Comment by queengooborg** [comment]
Odd, the CI should've failed because `package-lock.json` wasn't updated...  Could you update `package-lock.json` real quickly as well?

**Comment by saschanaz** [comment]
Could you do that for me, I'm on bed 🤔

**Review by queengooborg — APPROVED** [review]
Done, thank you!  This is LGTM, and essential for some plans Open Web Docs has, so I'll merge this now!

**Comment by saschanaz** [comment]
What plans, BTW? 👀

**Pull request merged**